### DISCUSSION TITLE
EntityFramework - Mapeamento de Entidades (Chave Primaria)

### DISCUSSION THREAD

**Discussion by FBODeveloper** [balta-io/forum#623] [Tira Dúvidas] [answered]
Pessoal...
Tenho uma entidade chamada Empresas dentro dela tenho id_empresa e id_escritorio, na hora de mapear como faço para que a chave primaria seja:
....
builder.HasKey(x => x.id_empresa, x.id_escritorio)

não consigo fazer isso com dois campos?

valeu
obrigado

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @FBODeveloper, como vai?

Para criar chaves compostas você pode definir um objeto anônimo e especificar as chaves:
```csharp
builder.HasKey(new { x.id_empresa, x.id_escritorio });
```

**Reply by FBODeveloper** [reply]
Valeu Balta..... consegui... ficou assim:
builder.HasKey(x => new { x.id_empresa, x.id_escritorio });

**Reply by andrebaltieri** [reply]
Showw 🚀
